datalake/src/main/java/com/sequenceiq/datalake/service/sdx/ProvisionerService.java

```diff
@@ -74,8 +74,13 @@ public void startStackDeletion(Long id, boolean forced) {
         SdxCluster sdxCluster = sdxService.getById(id);
         try {
             String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
-            ThreadBasedUserCrnProvider.doAsInternalActor(() ->
-                    stackV4Endpoint.deleteInternal(0L, sdxCluster.getClusterName(), forced, initiatorUserCrn));
+            if (sdxCluster.isDetached()) {
+                ThreadBasedUserCrnProvider.doAsInternalActor(() ->
+                        stackV4Endpoint.deleteDetached(0L, sdxCluster.getClusterName(), forced, initiatorUserCrn));
+            } else {
+                ThreadBasedUserCrnProvider.doAsInternalActor(() ->
+                        stackV4Endpoint.deleteInternal(0L, sdxCluster.getClusterName(), forced, initiatorUserCrn));
```

Comment:
This seems a little strange. I'd hope that delete internal could be able to realize that is was supposed to delete a detached cluster